### PR TITLE
Focus types

### PR DESCRIPTION
**Pull request by jtklein**
I added some of the recent updates to the Camera views into Seek and adapted some of the TypeScript code from those changes.
I am presenting the updates here as well.
Let me know what you think of it @albullington.

### CODE REVIEW

File: src/components/Camera/CameraView.tsx [hunk 1]
```diff
@@ -25,8 +24,8 @@ Reanimated.addWhitelistedNativeProps( {
 
 interface Props {
   animatedProps: CameraProps,
-  cameraRef: Object,
```
Comment (jtklein) [inline src/components/Camera/CameraView.tsx:28]: Those are more precise types available by the used libraries.

File: src/components/Camera/FocusSquare.tsx [hunk 2]
```diff
@@ -1,14 +1,13 @@
...
 import _ from "lodash";
-import type { Node } from "react";
 import React from "react";
 import { Animated } from "react-native";
 
 interface Props {
   animatedStyle: Object
 }
 
-const FocusSquare = ( { animatedStyle }: Props ): Node => {
```
Comment (jtklein) [inline src/components/Camera/FocusSquare.tsx:11]: Functional React components work well with inferred types: https://react-typescript-cheatsheet.netlify.app/docs/basic/getting-started/function_components

File: src/components/Camera/hooks/useFocusTap.ts [hunk 3]
```diff
@@ -20,7 +29,8 @@ const useFocusTap = ( cameraRef: Object, supportsFocus: boolean ): Node => {
     } );
   }, [tappedCoordinates, focusOpacity] );
 
-  const onFocus = useCallback( async ( { x, y }: GestureResponderEvent ) => {
+  type TapEvent = GestureStateChangeEvent<TapGestureHandlerEventPayload>;
```
Comment (jtklein) [inline src/components/Camera/hooks/useFocusTap.ts:32]: Gesture-handler also have their own types. I am not sure if this is necessary but in any case `GestureResponderEvent` is not exported from gesture-handler.

File: src/components/Camera/FocusSquare.tsx (hunk 2 above)
Comment (albullington) [reply src/components/Camera/FocusSquare.tsx:11]: It sounds like we can remove this Node type from all TS files going forward, yeah?

**Review by albullington — APPROVED** [review]
Looks good -- it's helpful to see what types we should be using for refs & that there are newer types available for gesture handler (I was looking at some pretty old examples in the react-native-vision-camera library).

File: src/components/Camera/FocusSquare.tsx (hunk 2 above)
Comment (jtklein) [reply src/components/Camera/FocusSquare.tsx:11]: Yes

**Pull request merged**